src/simulation/onboardDataHandling/storageUnit/_UnitTest/test_simpleStorageUnit.py

```diff
@@ -90,38 +90,150 @@ def checkStorage_limits(show_plots):
+    dat1Msg = messaging.DataNodeUsageMsg().write(dataMsg1)
 
-    if testFailCount:
-        print(testMessages)
-    else:
-        print("Passed")
+    dataMsg2 = messaging.DataNodeUsageMsgPayload()
+    dataMsg2.baudRate = baudRate_2 # baud
+    dataMsg2.dataName = "node_2_msg"
+    dat2Msg = messaging.DataNodeUsageMsg().write(dataMsg2)
 
-    return [testFailCount, ''.join(testMessages)]
+    # Test the addNodeToStorage method:
+    test_storage_unit.addDataNodeToModel(dat1Msg)
+    test_storage_unit.addDataNodeToModel(dat2Msg)
+
+    unitTestSim.AddModelToTask(unitTaskName, test_storage_unit)
+
+    dataLog = test_storage_unit.storageUnitDataOutMsg.recorder()
+    unitTestSim.AddModelToTask(unitTaskName, dataLog)
+    test_storage_unit.setDataBuffer(0)
+
+    unitTestSim.InitializeSimulation()
+    sim_time = 5.0
+    unitTestSim.ConfigureStopTime(macros.sec2nano(sim_time - 1.0))
+    unitTestSim.ExecuteSimulation()
+
+    test_storage_unit.setDataBuffer(add_data)
+    unitTestSim.ConfigureStopTime(macros.sec2nano(sim_time))
+    unitTestSim.ExecuteSimulation()
+
+    storedDataLog = dataLog.storageLevel
+    capacityLog = dataLog.storageCapacity
+    netBaudLog = dataLog.currentNetBaud
+    partitionName = dataLog.storedDataName
+    partitionData = dataLog.storedData
+
+    #   Check 1 - is net baud rate correct?
+    for ind in range(0,len(netBaudLog)):
+        currentBaud = netBaudLog[ind]
+        np.testing.assert_allclose(currentBaud, baudRate_1 + baudRate_2, atol=1e-4,
+            err_msg=("FAILED: PartitionedStorageUnit did not correctly log baud rate."))
+
+    #   Check 2 - is used storage space correct?
+    np.testing.assert_allclose(storedDataLog[-1], expectedStorage, atol=1e-4,
+        err_msg=("FAILED: PartitionedStorageUnit did not track integrated data."))
+
+    #   Check 3 - is the amount of data more than zero and less than the capacity?
+    for ind in range(0,len(storedDataLog)):
+        assert storedDataLog[ind] <= capacityLog[ind] or np.isclose(storedDataLog[ind], 
+            capacityLog[ind]), (
+            "FAILED: PartitionedStorageUnit's stored data exceeded its capacity.")
+
+        assert storedDataLog[ind] >= 0., (
+            "FAILED: PartitionedStorageUnit's stored data was negative.")
+
+    #   Check 4 - is the data in the partitioned storage unit correct?
+    assert partitionData[-1][0] == storedDataLog[-1], (
+        "FAILED: PartitionedStorageUnit did not correctly log the stored data.")
+    
+    #   Check 5 - is there only one partition?
+    assert len(partitionName[0]) == 1, (
+        "FAILED: PartitionedStorageUnit should have just one partition.")
+   
+    #   Check 6 - is the name of the partition correct?
+    assert partitionName[0][0] == "STORED DATA", (
+        "FAILED: PartitionedStorageUnit did not correctly log the partition name.")
 
 
 if __name__ == "__main__":
-    print(checkStorage_limits(False))
\ No newline at end of file
+    baudRate_1 = 1200
```

Comment:
I like the refactor to assert, np.testing asserts, and the significant addition of tests.